src/xb-value-bindings.c

```diff
@@ -17,30 +17,23 @@
 #include <glib.h>
 
 #include "xb-opcode-private.h"
+#include "xb-silo-private.h"
 #include "xb-value-bindings-private.h"
 
 typedef struct {
-	enum {
-		KIND_NONE,
-		KIND_TEXT,
-		KIND_INTEGER,
-	} kind;
-	union {
-		gchar *text;
-		guint32 integer;
-	};
+	XbOpcodeKind kind;
```

Comment:
Using `XbOpcodeKind` here seems risky, given it has a domain of values much wider than what’s needed here. There’s now no protection in the type system from ending up with `XB_OPCODE_KIND_INTEGER` rather than `XB_OPCODE_KIND_BOUND_INTEGER`, for example.

Reply:
Agreed, I've switched it back.